.codegen/model.go.tmpl

```diff
@@ -31,7 +31,7 @@ type {{.PascalName}} struct {
 {{end}}
 
 {{- define "field-tag" -}}
-	{{if .IsJson}}tfsdk:"{{if and (ne .Entity.Terraform nil) (ne .Entity.Terraform.Alias "") }}{{.Entity.Terraform.Alias}}{{else}}{{.Name}}{{end}}" tf:"{{if not .Required}}optional{{end}}"{{else}}tfsdk:"-"{{end -}}
+  {{if .IsJson}}tfsdk:"{{if and (ne .Entity.Terraform nil) (ne .Entity.Terraform.Alias "") }}{{.Entity.Terraform.Alias}}{{else}}{{.Name}}{{end}}" tf:"{{- $first := true -}}{{- if not .Required -}}{{- if not $first -}},{{end}}optional{{- $first = false -}}{{- end -}}{{- if .Entity.IsObject -}}{{- if not $first -}},{{end}}object{{- $first = false -}}{{- end -}}"{{else}}tfsdk:"-"{{end -}}
 {{- end -}}
```

Comment:
This is kinda ugly, mainly because I need to keep track of whether the item is the first one in the `tf` tags to add commas in the proper positions

Reply:
@hectorcast-db wondering if there's a better way

Reply:
You can create a function in the Go code to return the tfsdk options already formatted. So you can have all this logic in Go, which is more readable.

{{if .IsJson}}tfsdk:".Entity.TFSDKOptions" tf:".Entify.TFOptions"{{end-}}

Reply:
Sounds good, merging this PR for now to unblock and I'll work on the go function as a follow up